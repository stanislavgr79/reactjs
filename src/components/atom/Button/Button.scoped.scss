.btn {
    border-radius: 5px;
    font-size: 20px;
    border-style: none;
}

.btn_addMovie {
    color: #F65261;
    padding: 0.375rem 0.876rem;
    background: rgba(85, 85, 85, 0.5);
    font-weight: 600;
    &:hover {
        color: #FFFFFF;
    }
}

.btn_search {
    color: #FFFFFF;
    background-color: #F65261;
    font-weight: 500;
    padding: 0.475rem 3.476rem;
}

.btn_reset {
    color: #F65261;
    background: none;
    font-weight: 600;
    padding: 0.375rem 3.016rem;
    border: 2px solid #F65261;
    margin-right: 10px;
    &:hover {
        color: #FFFFFF;
    }
}

.btn_submit {
    color: #FFFFFF;
    background-color: #F65261;
    font-weight: 600;
    padding: 0.375rem 2.476rem;
    border: 2px solid #F65261;
    &:hover {
        color: #232323;
    }
}

.btn_close {
    color: #FFFFFF;
    background-image: url('../../../resources/images/icon_close_0.png');
    background-repeat: no-repeat;
    padding: 1rem 1rem;
}

.btn.btn_arrow {
    background-repeat: no-repeat;
    padding: 0.6rem 0.7rem;
    background-size: cover;
    margin-bottom: .5rem;
    border: none;
    box-shadow: none;
}

.arrow_up {
    background-image: url('../../../resources/images/arrow_up.png');
}

.arrow_down {
    background-image: url('../../../resources/images/arrow_down.png');
}