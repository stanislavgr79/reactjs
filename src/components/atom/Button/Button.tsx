import React from 'react';
import { createClassList } from '../../../helpers/utils';
import './Button.scoped.scss';

interface IProps {
  className?: string;
  id?: string;
  modifier?: string;
  onClick?: (e: unknown) => void;
  buttonType?: 'button' | 'submit' | 'reset';
  label?: string;
  sharingType?: string;
}

export default function Button({
  onClick,
  label = '',
  buttonType,
  className,
  id,
  sharingType,
}: IProps): JSX.Element {
  const classList = createClassList(['btn button', className || '']);

  return (
    <button
      className={classList}
      id={id || ''}
      onClick={onClick}
      type={buttonType || 'button'}
      data-type={sharingType}
    >
      {label}
    </button>
  );
}
