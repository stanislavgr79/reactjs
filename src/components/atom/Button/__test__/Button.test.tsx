import React from 'react';
import { fireEvent, render } from '@testing-library/react';

import Button from '../Button';

jest.mock('../../../../helpers/utils', () => ({
  createClassList: () => {
    return 'array of classes';
  },
}));

describe('Button', () => {
  const click = jest.fn();

  test('render', () => {
    const tree = render(
      <Button
        onClick={click}
        label="test-label"
        buttonType="submit"
        className={'test_class'}
        id={'test-id'}
        title={'test-title'}
        sharingType={'test-sharing'}
        disabled={false}
      />,
    );
    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render with out `type`, it must be `button`', () => {
    const tree = render(
      <Button
        onClick={click}
        label="test-label"
        className={'test_class'}
        id={'test-id'}
        title={'test-title'}
        sharingType={'test-sharing'}
        disabled={false}
      />,
    );
    expect(tree.baseElement).toMatchSnapshot();
  });

  test('render with out `className`', () => {
    const tree = render(
      <Button
        onClick={click}
        label="test-label"
        id={'test-id'}
        title={'test-title'}
        sharingType={'test-sharing'}
        disabled={false}
      />,
    );
    expect(tree.baseElement).toMatchSnapshot();
  });

  test('onClick', () => {
    const tree = render(<Button onClick={click} />);
    fireEvent.click(tree.getByRole('button'));
    expect(click).toBeCalled();
  });
});
