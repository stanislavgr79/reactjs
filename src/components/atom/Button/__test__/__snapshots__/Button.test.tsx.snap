// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render 1`] = `
<body>
  <div>
    <button
      class="array of classes"
      data-type="test-sharing"
      id="test-id"
      title="test-title"
      type="submit"
    >
      test-label
    </button>
  </div>
</body>
`;

exports[`Button render with out \`className\` 1`] = `
<body>
  <div>
    <button
      class="array of classes"
      data-type="test-sharing"
      id="test-id"
      title="test-title"
      type="button"
    >
      test-label
    </button>
  </div>
</body>
`;

exports[`Button render with out \`type\`, it must be \`button\` 1`] = `
<body>
  <div>
    <button
      class="array of classes"
      data-type="test-sharing"
      id="test-id"
      title="test-title"
      type="button"
    >
      test-label
    </button>
  </div>
</body>
`;
